### PR TITLE
feat(jitter): Separate jitter processing into a standalone package

### PR DESCRIPTION
**Pull request by tomtwinkle**
## Overview

This PR extracts the Jitter calculation logic, which was previously embedded within the `backoff` package, into a completely independent `jitter` package.
This decouples random number generation from the `backoff` package and allows the jitter logic to be reused generically in other contexts, such as simple polling loops.

Along with the extraction, this PR resolves built-in function shadowing, adds fail-safes for edge cases, implements tests, and expands the documentation.

## Changes

**Package Extraction**
- Created the new `jitter` package and implemented the `Apply` function.
- Moved the dependency on `math/rand/v2` from the `backoff` package to the `jitter` package.

**Resolved Built-in Function Shadowing**
- Renamed the argument/variable from `cap` to `duration` (in the `jitter` package) and `limit` (in the `backoff` package) to prevent shadowing the Go built-in `cap()` function. This eliminates compilation errors and potential bugs.

**Edge Case Handling & Safety Improvements**
- Added a fail-safe: if `duration` is `0` or negative, the function immediately returns `0`.
- Strictly defined boundary behaviors for the `factor` parameter (e.g., handling values `<= 0.0` or `> 1.0`).

**Testing Additions & Improvements**
- Implemented tests using exclusively the standard `testing` package, avoiding external dependencies like `testify`.
- Enabled parallel test execution using `t.Parallel()`.
- Added a stress test to validate Goroutine safety, utilizing Go 1.26's `t.Context()` to gracefully manage Goroutine lifecycles and context cancellations.

 **Documentation Updates**
- Created a dedicated `README.md` for the `jitter` package, detailing the behavior of Full Jitter vs. Partial Jitter, providing integration examples for custom polling, and explicitly documenting edge-case handling.
- Updated the `backoff` package's `README.md` to reflect the variable name change (`cap` to `limit`).

## Checklist

* [x] Verified that `go test -v -race ./...` passes successfully.
* [x] Confirmed no issues are reported by `go vet ./...`.
* [x] Ensured compilation errors caused by variable name shadowing are completely resolved.

### CODE REVIEW

File: jitter/jitter.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Package jitter provides functions for adding randomized jitter to time durations.
+// This is commonly used in retry mechanisms (like Exponential Backoff) to prevent
+// thundering-herd problems by spreading out the execution times of concurrent clients.
+package jitter
+
+import (
+	"math/rand/v2"
```
[the remark this answers is not in the document]
Comment (tomtwinkle) [reply jitter/jitter.go:7]: `math/rand/v2` is sufficiently secure because it uses ChaCha8. Furthermore, the jittering process does not require the use of high-precision random number generation like that provided by `crypto/rand`.

**Review by bushiyama — APPROVED** [review]
LGTM
